dydx/dydxPresenters/dydxPresenters/_v4/Vault/DepositsAndWithdrawals/dydxVaultDepositWithdrawConfirmationViewBuilder.swift

```diff
@@ -52,131 +58,189 @@ private protocol dydxVaultDepositWithdrawConfirmationViewPresenterProtocol: Host
+            .sink(receiveValue: { [weak self] vaultAndSubaccount, hasAcknowledgedHighSlippage in
+                guard self?.viewModel?.submitState != .submitting else { return }
+                self?.update(subaccount: vaultAndSubaccount.subaccount, vault: vaultAndSubaccount.vault, hasAcknowledgedHighSlippage: hasAcknowledgedHighSlippage)
+            })
             .store(in: &subscriptions)
-
-        AbacusStateManager.shared.state.selectedSubaccount
-            .sink { [weak self] selectedSubaccount in
-                self?.update(subaccount: selectedSubaccount)
+        
+        // need a non-debounce for the initial fetch so that UI updates immediately after change
+        viewModel.$hasAcknowledgedHighSlippage
+            .sink {[weak self] hasAcknowledgedHighSlippage in
+                guard let self = self else { return }
+                self.update(hasAcknowledgedHighSlippage: hasAcknowledgedHighSlippage)
             }
             .store(in: &subscriptions)
     }
-
+    
     private func initializeSubmitState() {
         guard let viewModel, let transferType else { return }
         switch transferType {
         case .deposit:
             viewModel.submitState = .enabled
         case .withdraw:
             viewModel.submitState = .loading
-            DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
-                // TO-DO replace fetch actual slippage and update view model
-                let slippage = Double.random(in: 0...0.02) // replace random with actual slippage
-                let requiresAcknowledgeHighSlippage = transferType == .withdraw && slippage >= Self.slippageAcknowledgementThreshold
-                viewModel.requiresAcknowledgeHighSlippage = requiresAcknowledgeHighSlippage
-                viewModel.slippage = slippage
-                if requiresAcknowledgeHighSlippage && !viewModel.hasAcknowledgedHighSlippage {
-                    self.viewModel?.submitState = .disabled
-                } else {
-                    self.viewModel?.submitState = .enabled
-                }
-            }
         }
     }
-
-    // TODO: replace with real data from abacus
-    func update(subaccount: Subaccount?) {
-        guard let amount,
-              let transferType,
-              amount > 0,
-              // TODO: replace
-              let curVaultBalance = Optional(420.0),
-              let curFreeCollateral = subaccount?.freeCollateral?.current?.doubleValue,
-              let curMarginUsage = subaccount?.marginUsage?.current?.doubleValue
-        else {
-//            assertionFailure()
+    
+    private func update(hasAcknowledgedHighSlippage: Bool) {
+        switch self.transferType {
+        case .deposit, nil:
+            // not applicable for deposits
+            break
+        case .withdraw:
+            self.viewModel?.submitState = .loading
+        }
+    }
+    
+    func update(subaccount: Subaccount?, vault: Abacus.Vault, hasAcknowledgedHighSlippage: Bool) {
```

Comment:
Nit: private?

Reply:
addressed in 9876fb38dbf9f612508276d203dc847a8b8b028b